apps/condo/domains/common/hooks/usePreviousQueryParams.ts

```diff
@@ -0,0 +1,149 @@
+
+        const pageKey = this.getPageKey(path, delimiter)
+
+        const stateFromLocalStorage = this.getAll()
+        const allQueryParamsByEmployeeId = get(stateFromLocalStorage, employeeId)
+        const newQueryParamsByEmployeeId = { ...allQueryParamsByEmployeeId, [pageKey]: queryParams }
+        const newStateAsJson = JSON.stringify({ ...stateFromLocalStorage, [employeeId]: newQueryParamsByEmployeeId })
+
+        localStorage.setItem(this.getStorageKey(), newStateAsJson)
+    }
+
+    static get (employeeId: string, path: string, delimiter?: Record<string, unknown>) {
+        if (!isString(employeeId) || !isString(path)) return
+
+        const pageKey = this.getPageKey(path, delimiter)
+
+        const stateAsJson = localStorage.getItem(this.getStorageKey())
+        if (!stateAsJson) return
+
+        try {
+            const stateFromLocalStorage = JSON.parse(stateAsJson)
+            if (!isObject(stateFromLocalStorage)) return
+
+            const allQueryParamsByOrganizationId = get(stateFromLocalStorage, employeeId)
+            if (!isObject(allQueryParamsByOrganizationId)) return
+
+            const queryParamsByPageKey = get(allQueryParamsByOrganizationId, pageKey)
+            if (!isObject(queryParamsByPageKey)) return
+
+            return queryParamsByPageKey
+        } catch (error) {
+            console.error('Cannot parse previous query params by path', { error, stateAsJson, pageKey, path, delimiter, employeeId })
+            return
+        }
+    }
+
+    private static getAll () {
+        const stateAsJson = localStorage.getItem(this.getStorageKey())
+        if (!stateAsJson) return
+
+        try {
+            const stateFromLocalStorage = JSON.parse(stateAsJson)
+            if (!isObject(stateFromLocalStorage)) return
+
+            return stateFromLocalStorage
+        } catch (error) {
+            console.error('Cannot parse previous query params', { error, stateAsJson })
+            return
+        }
+    }
+
+    private static getStorageKey () {
+        return `${this.STORAGE_KEY_PREFIX}-by-employee-id`
+    }
+
+    private static getPageKey (path: string, delimiter?: Record<string, unknown>) {
+        let pageKey = path
+        if (isObject(delimiter) && !isEmpty(delimiter)) {
+            const delimiterQueryParams = toPairs(delimiter)
+                .sort(([keyA], [keyB]) => keyA.localeCompare(keyB))
+                .map((paramKeyAndValue) => paramKeyAndValue.join('='))
+                .join('&')
+            if (delimiterQueryParams) pageKey += `?${delimiterQueryParams}`
+        }
+        return pageKey
+    }
+}
+
+const getParams = (query, paramsForSave) => {
+    return pick(query, paramsForSave)
+}
+
+type usePreviousQueryParamsType = (props: { trackedParamNames: Array<string>, delimitersParamNames?: Array<string>, employeeId?: string }) => void
+
+export const usePreviousQueryParams: usePreviousQueryParamsType = ({
+    delimitersParamNames,
```

Comment:
The name of the prop makes no sense.
Mb `tab` or some other name?

Reply:
I can't think of a good name for this variable. Changed it to `paramNamesForPageChange` and added a docs